magma/array.py

```diff
@@ -12,6 +12,7 @@
 from .debug import debug_wire, get_callee_frame_info
 from .logging import root_logger
 from magma.wire_container import WiringLog
+from magma.protocol_type import MagmaProtocol
```

Comment:
changes to this file and similar changes to tuple.py are for allowing protocol-type arguments to `Array` and `Tuple`.